.github/workflows/build-test-macos.yml

```diff
@@ -1,4 +1,4 @@
-name: build-and-test
+name: build-and-test-macos
```

Comment:
We only need one workflow that uploads test coverage. Can you remove the upload step from this workflow?

Reply:
This should be ok now

Reply:
From what I can remember, we do need coverage uploads for each OS because some tests don't run on every runner

Reply:
Is there no conflict/overriding that happens when we have multiple coverages being published for the same PR? I'll have to look into that again.

Reply:
Read up on Codecov and there's no issue: https://docs.codecov.com/docs/merging-reports We should upload coverage from all of the workflows